src/plugins/shape-generator/index.ts

```diff
@@ -0,0 +1,407 @@
+
+          if (isNaN(x) || isNaN(y) || isNaN(rx) || isNaN(ry) || isNaN(angle)) {
+            return;
+          }
+
+          // TODO: HelperExpression is missing unobserve types
+          angleHelper.unobserve("numericValue");
+
+          // Create new empty expression to add the ellipse to
+          this.calc.controller.dispatch({
+            type: "new-expression",
+          });
+
+          // Generate ellipse LaTeX
+          const latex = ellipseLatex(x, y, rx, ry, angle);
+
+          // Set the expression
+          this.calc.setExpression({
+            ...expr,
+            id: this.calc.selectedExpressionId,
+            type: "expression",
+            latex,
+          });
+
+          this.cleanupExpressions();
+          this.isEditingShape = false;
+        });
+      },
+    },
+    {
+      ariaLabel: "Add rectangle",
+      label: "rectangle",
+      okBtnExprId: "shape-generator-rectangle",
+      handler: () => {
+        this.calc.setExpressions(rectangleGeneratorExpressions);
+      },
+      okBtnHandler: (expr) => {
+        const xHelper = this.calc.HelperExpression({
+          latex: "x_{rectangleGenerator}",
+        });
+        const yHelper = this.calc.HelperExpression({
+          latex: "y_{rectangleGenerator}",
+        });
+        const wHelper = this.calc.HelperExpression({
+          latex: "w_{rectangleGenerator}",
+        });
+        const hHelper = this.calc.HelperExpression({
+          latex: "h_{rectangleGenerator}",
+        });
+        const angleHelper = this.calc.HelperExpression({
+          latex: "A_{rectangleGenerator}",
+        });
+
+        angleHelper.observe("numericValue", () => {
+          const x = xHelper.numericValue;
+          const y = yHelper.numericValue;
+          const w = wHelper.numericValue;
+          const h = hHelper.numericValue;
+          const angle = angleHelper.numericValue;
+
+          if (isNaN(x) || isNaN(y) || isNaN(w) || isNaN(h) || isNaN(angle)) {
+            return;
+          }
+
+          // TODO: HelperExpression is missing unobserve types
+          angleHelper.unobserve("numericValue");
+
+          // Create new empty expression to add the rectangle to
+          this.calc.controller.dispatch({
+            type: "new-expression",
+          });
+
+          // Generate rectangle LaTeX
+          const latex = rectangleLatex(w, h, x, y, angle);
+
+          // Set the expression
+          this.calc.setExpression({
+            ...expr,
+            id: this.calc.selectedExpressionId,
+            type: "expression",
+            latex,
+          });
+
+          this.cleanupExpressions();
+          this.isEditingShape = false;
+        });
+      },
+    },
+  ];
+
+  private expressionsMutationObserver: MutationObserver | null = null;
+
+  afterEnable() {
+    // Construct stylesheet
+    const style = document.createElement("style");
+    style.textContent = `
+      /* Disable the save button while editing a shape */
```

Comment:
These styles should be moved to a .less file. I think you can just create `shape-generator.less`,  put the code in there, and `import './shape-generator.less` from this index.ts file.

If these styles shouldn't run when the shape-generator plugin is disabled because they break other functionality, then they shouldn't run when the shape-generator plugin is enabled either.

If they merely don't _need_ to run when the plugin is disabled, there's not a concern with moving it to a .less file since a few more styles doesn't really hurt performance.